packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -679,6 +679,19 @@ extern dec deserializeEmptyStringAsNull(target: ModelProperty, scope?: valueof s
  */
 extern dec responseAsBool(target: Operation, scope?: valueof string);
 
+/**
+ * Change the operation location in client. If the target client is not defined, use `string` to indicate the client name.
+ *
+ * @param source The operation to change location for.
+ * @param target The target `Namespace`, `Interface` or a string which could indicate the client.
+ * @param scope The language scope for this decorator
+ */
+extern dec clientLocation(
+  source: Operation,
+  target: Interface | Namespace | (valueof string),
+  scope?: valueof string
```

Comment:
I know this is already an established pattern. However, if we add an options bag to hold all optional params we can get 2 main benefits:

1. Easy to add new params without breaking consumers.
2. Provide named params ({scope: "csharp"}) or ({isFoo: true}), which improves readability.

No need to change, but wanted to bring up this option for consideration at some point in the future